metricbeat/helper/interfacer.go

```diff
@@ -1,3 +1,17 @@
+/**
+
+
+= MetricSeter
+
+== Timeout
+Each metricseter must implement on his side to make sure the timeout is followed.
+Otherwise it can lead to the issue that multiple calls for one metricset happen in parallel and pile up.
+
+The initial idea was to use a Cancel function, but it was not implemented as it would add complexity by requiring
+an identifier for each fetch call and it would have to be decide, which fetch call must be canceled.
```

Comment:
Since this will be in godoc, I'm not sure it's worth talking about the option we didn't take. Btw, I like more the timeout path, it feels like more solid ground to me.